pyro/distributions/transforms/affine_coupling.py

```diff
@@ -125,11 +128,86 @@ def log_abs_det_jacobian(self, x, y):
             log_scale = self._cached_log_scale
         else:
             x1 = x[..., :self.split_dim]
-            _, log_scale = self.hypernet(x1)
+            _, log_scale = self.nn(x1)
             log_scale = clamp_preserve_gradients(log_scale, self.log_scale_min_clip, self.log_scale_max_clip)
         return log_scale.sum(-1)
 
 
+@copy_docs_from(ConditionalTransformModule)
+class ConditionalAffineCoupling(ConditionalTransformModule):
+    """
+    An implementation of the affine coupling layer of RealNVP (Dinh et al., 2017) that conditions on an additional
+    context variable and uses the bijective transform,
+
+        :math:`\\mathbf{y}_{1:d} = \\mathbf{x}_{1:d}`
+        :math:`\\mathbf{y}_{(d+1):D} = \\mu + \\sigma\\odot\\mathbf{x}_{(d+1):D}`
+
+    where :math:`\\mathbf{x}` are the inputs, :math:`\\mathbf{y}` are the outputs, e.g. :math:`\\mathbf{x}_{1:d}`
+    represents the first :math:`d` elements of the inputs, and :math:`\\mu,\\sigma` are shift and translation
+    parameters calculated as the output of a function inputting only :math:`\\mathbf{x}_{1:d}`.
+
+    That is, the first :math:`d` components remain unchanged, and the subsequent :math:`D-d` are shifted and
+    translated by a function of the previous components.
+
+    Together with :class:`~pyro.distributions.ConditionalTransformedDistribution` this provides a way to create
+    richer variational approximations.
+
+    Example usage:
+
+    >>> from pyro.nn import DenseNN
+    >>> input_dim = 10
+    >>> split_dim = 6
+    >>> base_dist = dist.Normal(torch.zeros(input_dim), torch.ones(input_dim))
+    >>> hypernet = DenseNN(split_dim, [10*input_dim], [input_dim-split_dim, input_dim-split_dim])
+    >>> transform = AffineCoupling(split_dim, hypernet)
+    >>> pyro.module("my_transform", transform)  # doctest: +SKIP
```

Comment:
nit: tweak the example to reflect usage of `ConditionalAffineCoupling`?

Reply:
Done!